.github/dependabot.yml

```diff
@@ -4,3 +4,11 @@ updates:
   directory: "/"
   schedule:
       interval: "weekly"
+
+- package-ecosystem: gomod
+  directory: "/"
+  schedule:
+    interval: "weekly"
+  open-pull-requests-limit: 10
+  vulnerability-alerts:
+    enabled: true
```

Comment:
I wasn't 100% sure this field will work. the safest way is refs to https://docs.github.com/en/code-security/dependabot/dependabot-alerts/configuring-dependabot-alerts
(need to configure by maintainer or user who have permission to write of this repo

Reply:
Do you know how does it tell us about the CVE detection? 

I feel like we won't want the CVEs to be public knowledge until we can fix and release them, so I want to be sure that dependabot won't start opening public PRs/issues about CVEs if we enable this. If it will, then we probably shouldn't add it and will have to come up with another way to do CVE detection.

Reply:
I need to investigate about it. will share on slack when I get more context

Reply:
<img width="1335" alt="image" src="https://github.com/kubernetes-sigs/kustomize/assets/73322208/bfc674ca-477a-49bb-8f4e-f16a6add8743">

I use my personal repo and manually enable CVE detect, it seems like it will show up here if there's any CVEs. but I'm not quite sure if it's visible for non-maintainer.

Reply:
<img width="754" alt="Screenshot 2023-10-26 at 4 33 46 PM" src="https://github.com/kubernetes-sigs/kustomize/assets/16629434/1b032ff1-acfe-45bb-9780-d643dc764f08">

Looks like it is only visible to maintainers, so that's great! Thank you for looking into it

Reply:
can you review other part of this PR? I think it's cover all we need currently(but still use simple method, if we need more complex logic then we need to see)

Reply:
Will add this to my review list for next week!

Reply:
Just took a quick look. I think the rest of this PR LGTM, but I think another eye will be good. @antoooks or @koba1t would one of you be able to take a look too?